html/gui/js/libraries/PlotlyUtilities.js

```diff
@@ -161,10 +213,301 @@ function generateChartOptions(record, params) { // eslint-disable-line no-unused
+            }
+        }
+
+        // Observed inconsistency with margin when subtitle is one line long. Unsure of the cause.
+        if (lineBreakCount > 0) {
+            if (firstRender) {
+                update['margin.t'] = marginTop + (titleHeight * lineBreakCount);
+            } else if (subtitleUpdates.subtitleLineCount === 1) {
+                marginTop = subtitleUpdates.chartUpdates['margin.t'] - (titleHeight * lineBreakCount);
+            } else {
+                marginTop -= (titleHeight * lineBreakCount);
+            }
+            if (topCenter) {
+                if (subtitleUpdates.subtitleLineCount > 0) {
+                    marginTop += subtitleHeight + 5;
+                    update['legend.y'] -= 0.025;
+                    update['margin.t'] += chartDiv._fullLayout.legend._height + subtitleHeight;
+                } else {
+                    marginTop -= chartDiv._fullLayout.legend._height / 2;
+                }
+            }
+        }
+
+        if (topCenter && subtitleUpdates.subtitleLineCount === 0 && !isPie) {
+            extraShift += 15;
+        }
+
+        marginTop += extraShift;
+
+        const titleYShift = (marginTop + legendHeight) - titleHeight;
+
+        if (titleIndex !== -1) {
+            update[`annotations[${titleIndex}].yshift`] = subtitleUpdates.subtitleLineCount >= 3 ? titleYShift + 5 : titleYShift;
+        }
+
+        if (subtitleIndex !== -1) {
+            update[`annotations[${subtitleIndex}].yshift`] = titleYShift - (subtitleHeight * subtitleUpdates.subtitleLineCount);
+        }
+
+        const marginBottom = chartDiv._fullLayout._size.b;
+        let pieChartXShift = 0;
+        if (isPie) {
+            pieChartXShift = subtitleUpdates.subtitleLineCount > 0 ? 2 : 1;
+        }
+
+        const shiftYDown = marginBottom * -1;
+        const exportShift = isExport ? 30 : 0;
+        if (creditsIndex !== -1) {
+            update[`annotations[${creditsIndex}].yshift`] = shiftYDown;
+            update[`annotations[${creditsIndex}].xshift`] = marginRight - pieChartXShift - exportShift;
+        }
+        if (restrictedIndex !== -1) {
+           update[`annotations[${restrictedIndex}].yshift`] = shiftYDown;
+           update[`annotations[${restrictedIndex}].xshift`] = (marginLeft - pieChartXShift - exportShift) * -1;
+        }
+    }
+    return update;
+}
+/**
+ * Changes Plotly JS legend_click and legend_doubleclick functionality.
+ * Disables legend_doubleclick event. Overrides legend_click
+ * to correctly hide Std Err traces when hiding traces and adjust axis
+ * tickmode due to Plotly JS bug with custom ticktext.
+ *
+ * @param  {Object} chartDiv - Plotly JS chart div.
+ */
+function overrideLegendEvent(chartDiv) { // eslint-disable-line no-unused-vars
+    chartDiv.on('plotly_legendclick', (evt) => {
+        // First check if all traces are hidden.
+        // There is a bug with tick text manually set.
+        // We need set the tickmode to auto if so.
```

Comment:
```suggestion
        // We need to set the tickmode to auto if so.
```
Sorry, I typoed my suggestion last time.